### PR TITLE
 Allow 'before' hook to modify all options

### PR DESCRIPTION
**Pull request by dpikt**
Previously, the "before" hook could only modify options passed to `Fetch`. This change allows it to modify all options, like `successDataPath` and `onSuccess`.

Other changes:
- Use async/await
- Update tests to use actual Fetch option names
- Use more explicit mock options

Tests are passing, so this more style/code review than QA.

### CODE REVIEW

File: src/http/http.js [hunk 1]
```diff
@@ -68,40 +68,45 @@ import {
...
+
+  const {
+    onSuccess, 
+    onFailure,
+    camelizeResponse,
+    successDataPath,
+    failureDataPath,
+    endpoint: url,
+    fetchOptions,
+  } = parsedOptions
```
Comment (chawes13) [inline src/http/http.js]: Thoughts on destructuring this on line 77?

File: src/http/http.js [hunk 2]
```diff
@@ -68,40 +68,45 @@ import {
...
+
+  try {
```
Comment (chawes13) [inline src/http/http.js]: Shouldn't this include the `await composeMiddleware()` step?

Comment (dpikt) [reply src/http/http.js]: I don't think it'd work since `onFailure` comes from that step

File: src/http/http.js (hunk 1 above)
Comment (dpikt) [reply src/http/http.js]: I left the variable to make it potentially more readable but if you don't think it's helpful I could consolidate!

Comment (chawes13) [reply src/http/http.js]: Nope, that makes sense to me

File: src/http/http.js (hunk 2 above)
Comment (chawes13) [reply src/http/http.js]: What happens then if `composeMiddleware` fails?

Comment (dpikt) [reply src/http/http.js]: Error throws, onFailure isn't called.

Comment (chawes13) [reply src/http/http.js]: I guess throwing the identity of the error is the same as letting the thrown error go unhandled?

Comment (dpikt) [reply src/http/http.js]: Yep. That wouldn't happen in the `composeMiddleware` case though if that's what you're wondering.

**Pull request merged**